src/services/lnd/index.ts

```diff
@@ -103,14 +103,38 @@ export const LndService = (): ILightningService | LightningServiceError => {
         id: paymentHash,
       })
       const { is_confirmed, is_failed, payment } = result
-      return {
-        status: is_confirmed
-          ? PaymentStatus.Settled
-          : is_failed
-          ? PaymentStatus.Failed
-          : PaymentStatus.Pending,
-        roundedUpFee: payment ? toSats(payment.safe_fee) : toSats(0),
+
+      const status = is_confirmed
+        ? PaymentStatus.Settled
+        : is_failed
+        ? PaymentStatus.Failed
+        : PaymentStatus.Pending
+
+      let paymentLookup: LnPaymentLookup = {
+        amount: toSats(0),
+        createdAt: new Date(0),
+        confirmedAt: undefined,
+        destination: "" as Pubkey,
+        roundedUpFee: toSats(0),
+        secret: "" as PaymentSecret,
+        request: undefined,
+        status,
       }
+
+      if (payment) {
+        paymentLookup = Object.assign(payment, {
+          amount: toSats(payment.tokens),
+          createdAt: new Date(payment.created_at),
```

Comment:
Just double-checking that these properties (`created_at`, `destination`, `request`) existed for you. The type checker is complaining on my end and defines `payment` prop on the type as:
```
export type GetPaymentResult = {
  .
  .
  payment?: {
    confirmed_at: string;
    fee_mtokens: string;
    hops: { ... }[];
    id: string;
    mtokens: string;
    safe_fee: number;
    safe_tokens: number;
    secret: string;
    timeout: number;
    tokens: number;
  };
};
```